### ISSUE TITLE
Create a player class

### ISSUE THREAD

**Issue by ZayLong** [AkaiArtsStudio/PlayDateSpriteMover#1]
Player Class Requirements: 
- [x] Needs to be able to take in a sprite sheed and create its own animation objects
- [x] Needs to be able to receive input from the player
- [x] needs to be able to detect collisions on itself
- [x] needs to be able to send and receive events 
- [x] needs to be able to move around in all directions

**Comment by ZayLong** [comment]
### What kind of properties would this player class need? 

a spritesheet reference
hasInput
speed
collider
coinCount

### Which of these props would need to be set upon creation? (:new()) 
spritesheet
speed

**Comment by ZayLong** [comment]
### what methods are needed for functionality?
spriteSetup - everything needed to setup sprite
movementInput
collisionEvent

**Comment by ZayLong** [comment]
for movement using `sprite:moveBy() `isn't enough as it doesn't take collisions into account.

So we're going to have to update to `sprite:moveWithCollisions()` and figure out how we're gonna handle those collisions when we come to them.

probably going to have to implement groups too so we know what we want to do when group x collides with group y. 

for future reference when we make static actor base class, we will want to implement `sprite:checkCollisions(x, y)` which does the same thing as` :moveWithCollisions` but doesn't actually MOVE the sprite.

to recap, the player needs a collider rect defined.
then, Player class needs an `update()` func
put `moveWithCollisions()` in the update allowing to move via deltaTime
we can process the results of moveWithCollisions and check if we have any group collisions.

lastly input.

We can create an input handler object that defines callbacks for each input but then feeds it to playdate. just call
`playdate.inputHandlers.push()`

We want to do this instead of hardcoding the input into the player class so we can easily change focus of the inputs to other objects.
